include/wlr/types/wlr_relative_pointer_v1.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_RELATIVE_POINTER_V1_H
+#define WLR_TYPES_WLR_RELATIVE_POINTER_V1_H
+
+#include <wayland-server.h>
+
+
+/**
+ * This protocol specifies a set of interfaces used for making clients able to
+ * receive relative pointer events not obstructed by barriers (such as the
+ * monitor edge or other pointer barriers).
+ */
+
+
+/**
+ * A global interface used for getting the relative pointer object for a given
+ * pointer.
+ */
+
+struct wlr_relative_pointer_manager_v1 {
+	struct wl_list resources;
+	struct wl_global *global;
+
+	struct {
+		struct wl_signal destroy;
+		/**
+		 * returns struct wlr_relative_pointer_v1 *relative_pointer
+		 * called when a new relative_pointer is successfully created
+		 */
+		struct wl_signal new_relative_pointer;
+	} events;
+
+	struct wl_listener display_destroy_listener;
+
+	void *data;
+};
+
+
+/**
+ * A wp_relative_pointer object is an extension to the wl_pointer interface
+ * used for emitting relative pointer events. It shares the same focus as
+ * wl_pointer objects of the same seat and will only emit events when it has
+ * focus.
+ */
+
+struct wlr_relative_pointer_v1 {
+	struct wl_client *client;
+	struct wl_resource *resource;
+	struct wl_resource *pointer;
```

Comment:
This field exposes a generic resource. No need for this. Instead, this struct should have a reference to the seat.

Reply:
I think we actually need *pointer.  We need to send the frame to the client.  Is there a way to know this pointer from the seat?

Reply:
Yes. Something like `seat.pointer_state.resource`.

Reply:
Oh ok I found it in seat_client.  I didn't realize sending events sent frames to all pointers in its list.